jina/clients/python/__init__.py

```diff
@@ -143,14 +149,14 @@ def input_fn(self) -> Union[Iterator['jina_pb2.Document'], Iterator[bytes], Call
         else:
             raise BadClient('input_fn is empty or not set')
 
-    @input_fn.setter
-    def input_fn(self, bytes_gen: Union[Iterator['jina_pb2.Document'], Iterator[bytes], Callable]) -> None:
+    def _set_input_fn(self, bytes_gen: Union[Iterator['jina_pb2.Document'], Iterator[bytes], Callable]) -> None:
         if self._input_fn:
             self.logger.warning('input_fn is not empty, overrided')
         if hasattr(bytes_gen, '__call__'):
-            self._input_fn = bytes_gen()
+            input_fn = bytes_gen()  # type: ignore
```

Comment:
remove type ignore, I would rather not touch any logic in a PR about improving tye annotation.